File: 7-kyu/colored-hexes!/python/solution.py
def hex_color(codes):
    if len(codes) == 0:
        return 'black'
    red, green, blue = [int(code) for code in codes.split()]
    greatest = max(red, green, blue)
    if red == green == blue:
        if red > 0:
            return 'white'
        else:
            return 'black'
    elif red == green:
        if blue > red:
            return 'blue'
        return 'yellow'
    elif green == blue:
        if red > blue:
            return 'red'
        return 'cyan'
    elif red == blue:
        if green > blue:
            return 'green'
        return 'magenta'
    elif greatest == red:
        return 'red'
    elif greatest == green:
        return 'green'
    elif greatest == blue:
        return 'blue'

File: 7-kyu/colored-hexes!/javascript/solution.js
const hexColor = codes => {
  if (codes.length === 0) {
    return "black";
  }
  const [red, green, blue] = codes.split(" ").map(n => Number.parseInt(n, 10));
  if (red === green && green === blue) {
    return red === 0 ? "black" : "white";
  }
  if (red === blue && red > green) {
    return "magenta";
  } else if (green === red && green > blue) {
    return "yellow";
  } else if (blue === green && blue > red) {
    return "cyan";
  }
  const greatest = Math.max(red, green, blue);
  if (greatest === red) {
    return "red";
  } else if (greatest === green) {
    return "green";
  }
  return "blue";
};
